### ISSUE TITLE
[URGENT NEED] sbt build.sh is failing

### ISSUE THREAD

**Issue by anirbanroydas** [scylladb/scylla-migrator#8]
Hi the `build.sh` is failing for me. I am using the `master` branch. I am not much familiar with **`scala`** and its **build tool**  `sbt` at all. I come from a _golang_, _python_ background mostly, so would really appreciate a quick help here.

My Current spark cluster consists of `3 ec2 instances` running `ubuntu 18.04` and I followed each steps to setup the cluster etc. All of them are running okay. I installed spark **2.3.1** version as suggested in the **_README_** (well that is the only source of documentation I found for it, it was not mentioned in the couple of blog articles from scylldb as well)

I am stuck at the `scylla-migrator` step where the `build.sh` is failing.

**NOTE:** I haven't exclusively installed scala in the node, since I read the spark binary comes pre build with scala.

To install sbt I followed the following instruction
```
echo "deb https://dl.bintray.com/sbt/debian /" | sudo tee -a /etc/apt/sources.list.d/sbt.list
curl -sL "https://keyserver.ubuntu.com/pks/lookup?op=get&search=0x2EE0EA64E40A89B84B2DF73499E82A75642AC823" | sudo apt-key add
sudo apt-get update
sudo apt-get install sbt
```
From the source https://www.scala-sbt.org/1.x/docs/Installing-sbt-on-Linux.html

```
$ java --version
openjdk 11.0.4 2019-07-16
OpenJDK Runtime Environment (build 11.0.4+11-post-Ubuntu-1ubuntu218.04.3)
OpenJDK 64-Bit Server VM (build 11.0.4+11-post-Ubuntu-1ubuntu218.04.3, mixed mode, sharing)
```

```
$ javac --version
javac 11.0.4
```

```
$ sbt --version
sbt version in this project: 1.3.3
sbt script version: 1.3.3
```

**Stack Trace**
```
$ <scylla-migrator-folder> ./build.sh
+ git submodule update --init --recursive
+ TMPDIR=/home/ubuntu/scylla-migrator/tmpexec
+ mkdir -p /home/ubuntu/scylla-migrator/tmpexec
+ trap 'rm -rf /home/ubuntu/scylla-migrator/tmpexec' EXIT
+ pushd spark-cassandra-connector
~/scylla-migrator/spark-cassandra-connector ~/scylla-migrator
+ sbt -Djava.io.tmpdir=/home/ubuntu/scylla-migrator/tmpexec -Dscala-2.11=true assembly
[ERROR] Failed to construct terminal; falling back to unsupported
java.lang.NumberFormatException: For input string: "0x100"
	at java.base/java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
	at java.base/java.lang.Integer.parseInt(Integer.java:652)
	at java.base/java.lang.Integer.valueOf(Integer.java:983)
	at jline.internal.InfoCmp.parseInfoCmp(InfoCmp.java:59)
	at jline.UnixTerminal.parseInfoCmp(UnixTerminal.java:233)
	at jline.UnixTerminal.<init>(UnixTerminal.java:64)
	at jline.UnixTerminal.<init>(UnixTerminal.java:49)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
	at java.base/java.lang.Class.newInstance(Class.java:584)
	at jline.TerminalFactory.getFlavor(TerminalFactory.java:209)
	at jline.TerminalFactory.create(TerminalFactory.java:100)
	at jline.TerminalFactory.get(TerminalFactory.java:184)
	at jline.TerminalFactory.get(TerminalFactory.java:190)
	at sbt.ConsoleLogger$.ansiSupported(ConsoleLogger.scala:123)
	at sbt.ConsoleLogger$.<init>(ConsoleLogger.scala:117)
	at sbt.ConsoleLogger$.<clinit>(ConsoleLogger.scala)
	at sbt.GlobalLogging$.initial(GlobalLogging.scala:43)
	at sbt.StandardMain$.initialGlobalLogging(Main.scala:61)
	at sbt.StandardMain$.initialState(Main.scala:70)
	at sbt.xMain.run(Main.scala:29)
	at xsbt.boot.Launch$$anonfun$run$1.apply(Launch.scala:111)
	at xsbt.boot.Launch$.withContextLoader(Launch.scala:130)
	at xsbt.boot.Launch$.run(Launch.scala:111)
	at xsbt.boot.Launch$$anonfun$apply$1.apply(Launch.scala:37)
	at xsbt.boot.Launch$.launch(Launch.scala:119)
	at xsbt.boot.Launch$.apply(Launch.scala:20)
	at xsbt.boot.Boot$.runImpl(Boot.scala:56)
	at xsbt.boot.Boot$.main(Boot.scala:18)
	at xsbt.boot.Boot.main(Boot.scala)

[info] Loading project definition from /home/ubuntu/scylla-migrator/spark-cassandra-connector/project
[info] Updating {file:/home/ubuntu/scylla-migrator/spark-cassandra-connector/project/}spark-cassandra-connector-build...
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by sbt.ivyint.ErrorMessageAuthenticator$ (file:/home/ubuntu/.sbt/boot/scala-2.10.6/org.scala-sbt/sbt/0.13.12/ivy-0.13.12.jar) to field java.net.Authenticator.theAuthenticator
WARNING: Please consider reporting this to the maintainers of sbt.ivyint.ErrorMessageAuthenticator$
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
[info] Resolving org.typelevel#sbt-typelevel;0.3.1 ...
[info] Resolving com.typesafe.sbt#sbt-pgp;0.8.2 ...
[info] Resolving com.jsuereth#gpg-library_2.10;0.8.2 ...
[info] Resolving org.scala-lang#scala-library;2.10.6 ...
[info] Resolving org.bouncycastle#bcpg-jdk15on;1.49 ...
[info] Resolving org.bouncycastle#bcprov-jdk15on;1.49 ...
[info] Resolving net.databinder#dispatch-http_2.10;0.8.10 ...
[info] Resolving net.databinder#dispatch-core_2.10;0.8.10 ...
[info] Resolving org.apache.httpcomponents#httpclient;4.1.3 ...
[info] Resolving org.apache.httpcomponents#httpcore;4.1.4 ...
[info] Resolving commons-logging#commons-logging;1.1.1 ...
[info] Resolving commons-codec#commons-codec;1.4 ...
[info] Resolving net.databinder#dispatch-futures_2.10;0.8.10 ...
[info] Resolving org.scala-lang#scala-actors;2.10.6 ...
[info] Resolving com.github.gseitz#sbt-release;0.8.4 ...
[info] Resolving com.typesafe#sbt-mima-plugin;0.1.6 ...
[info] Resolving com.typesafe#mima-reporter_2.10;0.1.6 ...
[info] Resolving com.typesafe#mima-core_2.10;0.1.6 ...
[info] Resolving org.scala-lang#scala-compiler;2.10.6 ...
[info] Resolving org.scala-lang#scala-reflect;2.10.6 ...
[info] Resolving org.scala-lang#scala-swing;2.10.2 ...
[info] Resolving com.typesafe#config;1.0.0 ...
[info] Resolving com.eed3si9n#sbt-buildinfo;0.3.2 ...
[info] Resolving org.xerial.sbt#sbt-sonatype;0.2.1 ...
[info] Resolving org.apache.httpcomponents#httpclient;4.2.6 ...
[info] Resolving org.apache.httpcomponents#httpcore;4.2.5 ...
[info] Resolving commons-codec#commons-codec;1.6 ...
[info] Resolving com.typesafe.sbt#sbt-git;0.8.5 ...
[info] Resolving com.typesafe.sbt#sbt-git;0.8.5 ...
[warn] 	module not found: com.typesafe.sbt#sbt-git;0.8.5
[warn] ==== typesafe-ivy-releases: tried
[warn]   https://repo.typesafe.com/typesafe/ivy-releases/com.typesafe.sbt/sbt-git/scala_2.10/sbt_0.13/0.8.5/ivys/ivy.xml
[warn] ==== sbt-plugin-releases: tried
[warn]   https://repo.scala-sbt.org/scalasbt/sbt-plugin-releases/com.typesafe.sbt/sbt-git/scala_2.10/sbt_0.13/0.8.5/ivys/ivy.xml
[warn] ==== local: tried
[warn]   /home/ubuntu/.ivy2/local/com.typesafe.sbt/sbt-git/scala_2.10/sbt_0.13/0.8.5/ivys/ivy.xml
[warn] ==== public: tried
[warn]   https://repo1.maven.org/maven2/com/typesafe/sbt/sbt-git_2.10_0.13/0.8.5/sbt-git-0.8.5.pom
[warn] ==== local-preloaded-ivy: tried
[warn]   /home/ubuntu/.sbt/preloaded/com.typesafe.sbt/sbt-git/0.8.5/ivys/ivy.xml
[warn] ==== local-preloaded: tried
[warn]   file:////home/ubuntu/.sbt/preloaded/com/typesafe/sbt/sbt-git_2.10_0.13/0.8.5/sbt-git-0.8.5.pom
[warn] ==== sonatype-releases: tried
[warn]   https://oss.sonatype.org/content/repositories/releases/com/typesafe/sbt/sbt-git_2.10_0.13/0.8.5/sbt-git-0.8.5.pom
[warn] ==== sbt-plugin-releases: tried
[warn]   https://repo.scala-sbt.org/scalasbt/sbt-plugin-releases/com.typesafe.sbt/sbt-git/scala_2.10/sbt_0.13/0.8.5/ivys/ivy.xml
[warn] ==== public: tried
[warn]   https://repo1.maven.org/maven2/com/typesafe/sbt/sbt-git_2.10_0.13/0.8.5/sbt-git-0.8.5.pom
[warn] ==== sonatype-public: tried
[warn]   https://oss.sonatype.org/content/repositories/public/com/typesafe/sbt/sbt-git_2.10_0.13/0.8.5/sbt-git-0.8.5.pom
[warn] ==== typesafe-releases: tried
[warn]   https://repo.typesafe.com/typesafe/releases/com/typesafe/sbt/sbt-git_2.10_0.13/0.8.5/sbt-git-0.8.5.pom
[warn] ==== Spark Packages Main repo: tried
[warn]   https://dl.bintray.com/spark-packages/maven/com/typesafe/sbt/sbt-git_2.10_0.13/0.8.5/sbt-git-0.8.5.pom
[warn] ==== Spark Packages Main repo: tried
[warn]   https://dl.bintray.com/spark-packages/maven/com/typesafe/sbt/sbt-git_2.10_0.13/0.8.5/sbt-git-0.8.5.pom
[info] Resolving com.scalapenos#sbt-prompt;1.0.0 ...
[info] Resolving org.slf4j#slf4j-nop;1.7.21 ...
[info] Resolving org.slf4j#slf4j-api;1.7.21 ...
[info] Resolving org.scalastyle#scalastyle-sbt-plugin;0.8.0 ...
[info] Resolving org.scalastyle#scalastyle_2.10;0.8.0 ...
[info] Resolving org.scalariform#scalariform_2.10;0.1.7 ...
[info] Resolving com.typesafe#config;1.2.0 ...
[info] Resolving org.scoverage#sbt-scoverage;1.0.4 ...
[info] Resolving org.scoverage#scalac-scoverage-plugin_2.10;1.0.4 ...
[info] Resolving org.spark-packages#sbt-spark-package;0.2.5 ...
[info] Resolving com.eed3si9n#sbt-assembly;0.14.3 ...
[info] Resolving org.scalactic#scalactic_2.10;2.2.1 ...
[info] Resolving org.pantsbuild#jarjar;1.6.0 ...
[info] Resolving org.apache.ant#ant;1.9.6 ...
[info] Resolving org.apache.ant#ant-launcher;1.9.6 ...
[info] Resolving org.ow2.asm#asm;5.0.4 ...
[info] Resolving org.ow2.asm#asm-commons;5.0.4 ...
[info] Resolving org.ow2.asm#asm-tree;5.0.4 ...
[info] Resolving org.apache.maven#maven-plugin-api;3.3.3 ...
[info] Resolving org.apache.maven#maven-model;3.3.3 ...
[info] Resolving org.codehaus.plexus#plexus-utils;3.0.20 ...
[info] Resolving org.apache.maven#maven-artifact;3.3.3 ...
[info] Resolving org.eclipse.sisu#org.eclipse.sisu.plexus;0.3.0 ...
[info] Resolving javax.enterprise#cdi-api;1.0 ...
[info] Resolving javax.annotation#jsr250-api;1.0 ...
[info] Resolving javax.inject#javax.inject;1 ...
[info] Resolving org.eclipse.sisu#org.eclipse.sisu.inject;0.3.0 ...
[info] Resolving org.codehaus.plexus#plexus-component-annotations;1.5.5 ...
[info] Resolving org.codehaus.plexus#plexus-classworlds;2.5.2 ...
[info] Resolving org.scalaj#scalaj-http_2.10;1.1.4 ...
[info] Resolving org.apache.directory.studio#org.apache.commons.codec;1.8 ...
[info] Resolving commons-codec#commons-codec;1.8 ...
[info] Resolving net.virtual-void#sbt-dependency-graph;0.8.2 ...
[info] Resolving com.github.mdr#ascii-graphs_2.10;0.0.3 ...
[info] Resolving io.get-coursier#sbt-coursier;1.0.0 ...
[info] Resolving io.get-coursier#coursier_2.10;1.0.0 ...
[info] Resolving org.scalamacros#quasiquotes_2.10;2.1.0 ...
[info] Resolving org.scalaz#scalaz-core_2.10;7.2.16 ...
[info] Resolving io.get-coursier#coursier-cache_2.10;1.0.0 ...
[info] Resolving org.scalaz#scalaz-concurrent_2.10;7.2.16 ...
[info] Resolving org.scalaz#scalaz-effect_2.10;7.2.16 ...
[info] Resolving io.get-coursier#coursier-extra_2.10;1.0.0 ...
[info] Resolving io.get-coursier#sbt-shared;1.0.0 ...
[info] Resolving org.scala-sbt#sbt;0.13.12 ...
[info] Resolving org.scala-sbt#main;0.13.12 ...
[info] Resolving org.scala-sbt#actions;0.13.12 ...
[info] Resolving org.scala-sbt#classpath;0.13.12 ...
[info] Resolving org.scala-sbt#interface;0.13.12 ...
[info] Resolving org.scala-sbt#io;0.13.12 ...
[info] Resolving org.scala-sbt#control;0.13.12 ...
[info] Resolving org.scala-sbt#launcher-interface;1.0.0-M1 ...
[info] Resolving org.scala-sbt#completion;0.13.12 ...
[info] Resolving org.scala-sbt#collections;0.13.12 ...
[info] Resolving jline#jline;2.13 ...
[info] Resolving org.fusesource.jansi#jansi;1.11 ...
[info] Resolving org.scala-sbt#api;0.13.12 ...
[info] Resolving org.scala-sbt#classfile;0.13.12 ...
[info] Resolving org.scala-sbt#logging;0.13.12 ...
[info] Resolving org.scala-sbt#process;0.13.12 ...
[info] Resolving org.scala-sbt#compiler-integration;0.13.12 ...
[info] Resolving org.scala-sbt#incremental-compiler;0.13.12 ...
[info] Resolving org.scala-sbt#relation;0.13.12 ...
[info] Resolving org.scala-sbt#compile;0.13.12 ...
[info] Resolving org.scala-sbt#persist;0.13.12 ...
[info] Resolving org.scala-tools.sbinary#sbinary_2.10;0.4.2 ...
[info] Resolving org.scala-sbt#compiler-ivy-integration;0.13.12 ...
[info] Resolving org.scala-sbt#ivy;0.13.12 ...
[info] Resolving org.scala-sbt#cross;0.13.12 ...
[info] Resolving org.scala-sbt.ivy#ivy;2.3.0-sbt-2cc8d2761242b072cedb0a04cb39435c4fa24f9a ...
[info] Resolving com.jcraft#jsch;0.1.46 ...
[info] Resolving org.scala-sbt#serialization_2.10;0.1.2 ...
[info] Resolving org.scala-lang.modules#scala-pickling_2.10;0.10.1 ...
[info] Resolving org.scalamacros#quasiquotes_2.10;2.0.1 ...
[info] Resolving org.json4s#json4s-core_2.10;3.2.10 ...
[info] Resolving org.json4s#json4s-ast_2.10;3.2.10 ...
[info] Resolving com.thoughtworks.paranamer#paranamer;2.6 ...
[info] Resolving org.spire-math#jawn-parser_2.10;0.6.0 ...
[info] Resolving org.spire-math#json4s-support_2.10;0.6.0 ...
[info] Resolving org.scala-sbt#run;0.13.12 ...
[info] Resolving org.scala-sbt#task-system;0.13.12 ...
[info] Resolving org.scala-sbt#tasks;0.13.12 ...
[info] Resolving org.scala-sbt#tracking;0.13.12 ...
[info] Resolving org.scala-sbt#cache;0.13.12 ...
[info] Resolving org.scala-sbt#testing;0.13.12 ...
[info] Resolving org.scala-sbt#test-agent;0.13.12 ...
[info] Resolving org.scala-sbt#test-interface;1.0 ...
[info] Resolving org.scala-sbt#main-settings;0.13.12 ...
[info] Resolving org.scala-sbt#apply-macro;0.13.12 ...
[info] Resolving org.scala-sbt#command;0.13.12 ...
[info] Resolving org.scala-sbt#logic;0.13.12 ...
[info] Resolving org.scala-sbt#compiler-interface;0.13.12 ...
[info] Resolving org.scala-lang#jline;2.10.6 ...
[info] Resolving org.fusesource.jansi#jansi;1.4 ...
[warn] 	::::::::::::::::::::::::::::::::::::::::::::::
[warn] 	::          UNRESOLVED DEPENDENCIES         ::
[warn] 	::::::::::::::::::::::::::::::::::::::::::::::
[warn] 	:: com.typesafe.sbt#sbt-git;0.8.5: not found
[warn] 	::::::::::::::::::::::::::::::::::::::::::::::
[warn]
[warn] 	Note: Some unresolved dependencies have extra attributes.  Check that these dependencies exist with the requested attributes.
[warn] 		com.typesafe.sbt:sbt-git:0.8.5 (scalaVersion=2.10, sbtVersion=0.13)
[warn]
[warn] 	Note: Unresolved dependencies path:
[warn] 		com.typesafe.sbt:sbt-git:0.8.5 (scalaVersion=2.10, sbtVersion=0.13) (/home/ubuntu/scylla-migrator/spark-cassandra-connector/project/plugins.sbt#L17-18)
[warn] 		  +- com.scalapenos:sbt-prompt:1.0.0 (scalaVersion=2.10, sbtVersion=0.13) (/home/ubuntu/scylla-migrator/spark-cassandra-connector/project/plugins.sbt#L19-20)
[warn] 		  +- default:spark-cassandra-connector-build:0.1-SNAPSHOT (scalaVersion=2.10, sbtVersion=0.13)
sbt.ResolveException: unresolved dependency: com.typesafe.sbt#sbt-git;0.8.5: not found
	at sbt.IvyActions$.sbt$IvyActions$$resolve(IvyActions.scala:313)
	at sbt.IvyActions$$anonfun$updateEither$1.apply(IvyActions.scala:191)
	at sbt.IvyActions$$anonfun$updateEither$1.apply(IvyActions.scala:168)
	at sbt.IvySbt$Module$$anonfun$withModule$1.apply(Ivy.scala:156)
	at sbt.IvySbt$Module$$anonfun$withModule$1.apply(Ivy.scala:156)
	at sbt.IvySbt$$anonfun$withIvy$1.apply(Ivy.scala:133)
	at sbt.IvySbt.sbt$IvySbt$$action$1(Ivy.scala:57)
	at sbt.IvySbt$$anon$4.call(Ivy.scala:65)
	at xsbt.boot.Locks$GlobalLock.withChannel$1(Locks.scala:95)
	at xsbt.boot.Locks$GlobalLock.xsbt$boot$Locks$GlobalLock$$withChannelRetries$1(Locks.scala:80)
	at xsbt.boot.Locks$GlobalLock$$anonfun$withFileLock$1.apply(Locks.scala:99)
	at xsbt.boot.Using$.withResource(Using.scala:10)
	at xsbt.boot.Using$.apply(Using.scala:9)
	at xsbt.boot.Locks$GlobalLock.ignoringDeadlockAvoided(Locks.scala:60)
	at xsbt.boot.Locks$GlobalLock.withLock(Locks.scala:50)
	at xsbt.boot.Locks$.apply0(Locks.scala:31)
	at xsbt.boot.Locks$.apply(Locks.scala:28)
	at sbt.IvySbt.withDefaultLogger(Ivy.scala:65)
	at sbt.IvySbt.withIvy(Ivy.scala:128)
	at sbt.IvySbt.withIvy(Ivy.scala:125)
	at sbt.IvySbt$Module.withModule(Ivy.scala:156)
	at sbt.IvyActions$.updateEither(IvyActions.scala:168)
	at sbt.Classpaths$$anonfun$sbt$Classpaths$$work$1$1.apply(Defaults.scala:1442)
	at sbt.Classpaths$$anonfun$sbt$Classpaths$$work$1$1.apply(Defaults.scala:1438)
	at sbt.Classpaths$$anonfun$doWork$1$1$$anonfun$90.apply(Defaults.scala:1473)
	at sbt.Classpaths$$anonfun$doWork$1$1$$anonfun$90.apply(Defaults.scala:1471)
	at sbt.Tracked$$anonfun$lastOutput$1.apply(Tracked.scala:37)
	at sbt.Classpaths$$anonfun$doWork$1$1.apply(Defaults.scala:1476)
	at sbt.Classpaths$$anonfun$doWork$1$1.apply(Defaults.scala:1470)
	at sbt.Tracked$$anonfun$inputChanged$1.apply(Tracked.scala:60)
	at sbt.Classpaths$.cachedUpdate(Defaults.scala:1493)
	at sbt.Classpaths$$anonfun$updateTask$1.apply(Defaults.scala:1420)
	at sbt.Classpaths$$anonfun$updateTask$1.apply(Defaults.scala:1372)
	at scala.Function1$$anonfun$compose$1.apply(Function1.scala:47)
	at sbt.$tilde$greater$$anonfun$$u2219$1.apply(TypeFunctions.scala:40)
	at sbt.std.Transform$$anon$4.work(System.scala:63)
	at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
	at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
	at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
	at sbt.Execute.work(Execute.scala:237)
	at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
	at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
	at sbt.ConcurrentRestrictions$$anon$4$$anonfun$1.apply(ConcurrentRestrictions.scala:159)
	at sbt.CompletionService$$anon$2.call(CompletionService.scala:28)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
[error] (*:update) sbt.ResolveException: unresolved dependency: com.typesafe.sbt#sbt-git;0.8.5: not found
[ERROR] Failed to construct terminal; falling back to unsupported
java.lang.NumberFormatException: For input string: "0x100"
	at java.base/java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
	at java.base/java.lang.Integer.parseInt(Integer.java:652)
	at java.base/java.lang.Integer.valueOf(Integer.java:983)
	at jline.internal.InfoCmp.parseInfoCmp(InfoCmp.java:59)
	at jline.UnixTerminal.parseInfoCmp(UnixTerminal.java:233)
	at jline.UnixTerminal.<init>(UnixTerminal.java:64)
	at jline.UnixTerminal.<init>(UnixTerminal.java:49)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
	at java.base/java.lang.Class.newInstance(Class.java:584)
	at jline.TerminalFactory.getFlavor(TerminalFactory.java:209)
	at jline.TerminalFactory.create(TerminalFactory.java:100)
	at jline.TerminalFactory.get(TerminalFactory.java:184)
	at jline.TerminalFactory.get(TerminalFactory.java:190)
	at sbt.JLine$.sbt$JLine$$terminal(LineReader.scala:85)
	at sbt.JLine$.withTerminal(LineReader.scala:88)
	at sbt.JLine$.usingTerminal(LineReader.scala:96)
	at sbt.JLine$.createReader(LineReader.scala:102)
	at sbt.SimpleReader.<init>(LineReader.scala:139)
	at sbt.SimpleReader$.<init>(LineReader.scala:141)
	at sbt.SimpleReader$.<clinit>(LineReader.scala)
	at sbt.BuiltinCommands$.sbt$BuiltinCommands$$doLoadFailed(Main.scala:454)
	at sbt.BuiltinCommands$$anonfun$loadFailed$1.apply(Main.scala:446)
	at sbt.BuiltinCommands$$anonfun$loadFailed$1.apply(Main.scala:446)
	at sbt.Command$$anonfun$applyEffect$1$$anonfun$apply$2.apply(Command.scala:59)
	at sbt.Command$$anonfun$applyEffect$1$$anonfun$apply$2.apply(Command.scala:59)
	at sbt.Command$$anonfun$applyEffect$2$$anonfun$apply$3.apply(Command.scala:61)
	at sbt.Command$$anonfun$applyEffect$2$$anonfun$apply$3.apply(Command.scala:61)
	at sbt.Command$.process(Command.scala:93)
	at sbt.MainLoop$$anonfun$1$$anonfun$apply$1.apply(MainLoop.scala:96)
	at sbt.MainLoop$$anonfun$1$$anonfun$apply$1.apply(MainLoop.scala:96)
	at sbt.State$$anon$1.process(State.scala:184)
	at sbt.MainLoop$$anonfun$1.apply(MainLoop.scala:96)
	at sbt.MainLoop$$anonfun$1.apply(MainLoop.scala:96)
	at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
	at sbt.MainLoop$.next(MainLoop.scala:96)
	at sbt.MainLoop$.run(MainLoop.scala:89)
	at sbt.MainLoop$$anonfun$runWithNewLog$1.apply(MainLoop.scala:68)
	at sbt.MainLoop$$anonfun$runWithNewLog$1.apply(MainLoop.scala:63)
	at sbt.Using.apply(Using.scala:24)
	at sbt.MainLoop$.runWithNewLog(MainLoop.scala:63)
	at sbt.MainLoop$.runAndClearLast(MainLoop.scala:46)
	at sbt.MainLoop$.runLoggedLoop(MainLoop.scala:30)
	at sbt.MainLoop$.runLogged(MainLoop.scala:22)
	at sbt.StandardMain$.runManaged(Main.scala:54)
	at sbt.xMain.run(Main.scala:29)
	at xsbt.boot.Launch$$anonfun$run$1.apply(Launch.scala:111)
	at xsbt.boot.Launch$.withContextLoader(Launch.scala:130)
	at xsbt.boot.Launch$.run(Launch.scala:111)
	at xsbt.boot.Launch$$anonfun$apply$1.apply(Launch.scala:37)
	at xsbt.boot.Launch$.launch(Launch.scala:119)
	at xsbt.boot.Launch$.apply(Launch.scala:20)
	at xsbt.boot.Boot$.runImpl(Boot.scala:56)
	at xsbt.boot.Boot$.main(Boot.scala:18)
	at xsbt.boot.Boot.main(Boot.scala)

[ERROR] Failed to construct terminal; falling back to unsupported
java.lang.NumberFormatException: For input string: "0x100"
	at java.base/java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
	at java.base/java.lang.Integer.parseInt(Integer.java:652)
	at java.base/java.lang.Integer.valueOf(Integer.java:983)
	at jline.internal.InfoCmp.parseInfoCmp(InfoCmp.java:59)
	at jline.UnixTerminal.parseInfoCmp(UnixTerminal.java:233)
	at jline.UnixTerminal.<init>(UnixTerminal.java:64)
	at jline.UnixTerminal.<init>(UnixTerminal.java:49)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
	at java.base/java.lang.Class.newInstance(Class.java:584)
	at jline.TerminalFactory.getFlavor(TerminalFactory.java:209)
	at jline.TerminalFactory.create(TerminalFactory.java:100)
	at jline.TerminalFactory.get(TerminalFactory.java:184)
	at jline.TerminalFactory.get(TerminalFactory.java:190)
	at jline.console.ConsoleReader.<init>(ConsoleReader.java:240)
	at jline.console.ConsoleReader.<init>(ConsoleReader.java:232)
	at jline.console.ConsoleReader.<init>(ConsoleReader.java:220)
	at sbt.JLine$$anonfun$createReader$1.apply(LineReader.scala:103)
	at sbt.JLine$$anonfun$createReader$1.apply(LineReader.scala:102)
	at sbt.JLine$$anonfun$usingTerminal$1.apply(LineReader.scala:98)
	at sbt.JLine$$anonfun$usingTerminal$1.apply(LineReader.scala:96)
	at sbt.JLine$.withTerminal(LineReader.scala:89)
	at sbt.JLine$.usingTerminal(LineReader.scala:96)
	at sbt.JLine$.createReader(LineReader.scala:102)
	at sbt.SimpleReader.<init>(LineReader.scala:139)
	at sbt.SimpleReader$.<init>(LineReader.scala:141)
	at sbt.SimpleReader$.<clinit>(LineReader.scala)
	at sbt.BuiltinCommands$.sbt$BuiltinCommands$$doLoadFailed(Main.scala:454)
	at sbt.BuiltinCommands$$anonfun$loadFailed$1.apply(Main.scala:446)
	at sbt.BuiltinCommands$$anonfun$loadFailed$1.apply(Main.scala:446)
	at sbt.Command$$anonfun$applyEffect$1$$anonfun$apply$2.apply(Command.scala:59)
	at sbt.Command$$anonfun$applyEffect$1$$anonfun$apply$2.apply(Command.scala:59)
	at sbt.Command$$anonfun$applyEffect$2$$anonfun$apply$3.apply(Command.scala:61)
	at sbt.Command$$anonfun$applyEffect$2$$anonfun$apply$3.apply(Command.scala:61)
	at sbt.Command$.process(Command.scala:93)
	at sbt.MainLoop$$anonfun$1$$anonfun$apply$1.apply(MainLoop.scala:96)
	at sbt.MainLoop$$anonfun$1$$anonfun$apply$1.apply(MainLoop.scala:96)
	at sbt.State$$anon$1.process(State.scala:184)
	at sbt.MainLoop$$anonfun$1.apply(MainLoop.scala:96)
	at sbt.MainLoop$$anonfun$1.apply(MainLoop.scala:96)
	at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
	at sbt.MainLoop$.next(MainLoop.scala:96)
	at sbt.MainLoop$.run(MainLoop.scala:89)
	at sbt.MainLoop$$anonfun$runWithNewLog$1.apply(MainLoop.scala:68)
	at sbt.MainLoop$$anonfun$runWithNewLog$1.apply(MainLoop.scala:63)
	at sbt.Using.apply(Using.scala:24)
	at sbt.MainLoop$.runWithNewLog(MainLoop.scala:63)
	at sbt.MainLoop$.runAndClearLast(MainLoop.scala:46)
	at sbt.MainLoop$.runLoggedLoop(MainLoop.scala:30)
	at sbt.MainLoop$.runLogged(MainLoop.scala:22)
	at sbt.StandardMain$.runManaged(Main.scala:54)
	at sbt.xMain.run(Main.scala:29)
	at xsbt.boot.Launch$$anonfun$run$1.apply(Launch.scala:111)
	at xsbt.boot.Launch$.withContextLoader(Launch.scala:130)
	at xsbt.boot.Launch$.run(Launch.scala:111)
	at xsbt.boot.Launch$$anonfun$apply$1.apply(Launch.scala:37)
	at xsbt.boot.Launch$.launch(Launch.scala:119)
	at xsbt.boot.Launch$.apply(Launch.scala:20)
	at xsbt.boot.Boot$.runImpl(Boot.scala:56)
	at xsbt.boot.Boot$.main(Boot.scala:18)
	at xsbt.boot.Boot.main(Boot.scala)
```

**Comment by anirbanroydas** [comment]
Hi @iravid, tagging you here just get some quick resolution here.

**Comment by anirbanroydas** [comment]
@iravid Does it have to with my java version, looking into the `build.sbt` file, I found
`javacOptions ++= Seq("-source", "1.8", "-target", "1.8"),`
Which may seem to be different from my `javac --version` result?

**Comment by anirbanroydas** [comment]
@iravid I actually tested building it in my mac. With java 8 too and still it didn't work. Which tells me its not related to java probably and related to the missing dependency `com.typesafe.sbt#sbt-git;0.8.5: not found`

On mac java version
```
 $ java -version
openjdk version "1.8.0_232"
OpenJDK Runtime Environment (AdoptOpenJDK)(build 1.8.0_232-b09)
OpenJDK 64-Bit Server VM (AdoptOpenJDK)(build 25.232-b09, mixed mode)
```

I would really like some help here since I am trying out a scylla migration and looking for the spark based solution.

**Comment by tarzanek** [comment]
and I am quite sure you need java 1.8 (I actually built on Ubuntu stable with sbt 1.3.3 and jdk 1.8), let me send you my env settings, I think you're close to getting it built on your linux box

**Comment by anirbanroydas** [comment]
@tarzanek Okay appreciate your help. Let me know when you get to build it successfully.

I tried in ubuntu 18.04 and in mac os. Both didn't work for me. Since I am not really familiar with the build tools and I am not able to hack into it to make it work either.

**Comment by anirbanroydas** [comment]
Hi guys, would really appreciate some solution here. Kind of in a fast paced place needing the migration done. Did anyone figure out what is the issue with the above failing build?
@iravid  @tarzanek

**Comment by tarzanek** [comment]
```
➜  scylla-migrator git:(master) lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 19.10
Release:	19.10
Codename:	eoan

➜  scylla-migrator git:(master) javac -version
javac 1.8.0_232-ea
➜  scylla-migrator git:(master) javac -version
javac 1.8.0_232-ea

```
is my environment that can get you past the issue above

**Comment by anirbanroydas** [comment]
Thanks @tarzanek.  Yeah so I want to stick to 18.04. What patches needed to support it in 18.04. What is the exact issue that I am facing which is failing to build it? Any ideas on that?

**Comment by tarzanek** [comment]
I think 18.04 will work too with above versions of java and sbt, however I found another problem(merge issue in master), so let me figure it out

**Comment by tarzanek** [comment]
btw. in your above environment you should be able to also proceed by changing java to 1.8

**Comment by tarzanek** [comment]
```
➜  scylla-migrator git:(master) java -version
openjdk version "1.8.0_232"
OpenJDK Runtime Environment (build 1.8.0_232-b09)
OpenJDK 64-Bit Server VM (build 25.232-b09, mixed mode)

➜  scylla-migrator git:(master) sbt --version
sbt version in this project: 1.3.3
sbt script version: 1.3.3
```
on Fedora 30 also works, so this problem you saw is above jdk 1.8 for sure

**Comment by tarzanek** [comment]
also the other idea I have that there is something in your env that might change how sbt works?
can you check `env` ?

**Comment by tarzanek** [comment]
the other idea I have is that you might try to check sbt cache, I saw cases where the cache was corrupted and it broke dependency lookups

**Comment by tarzanek** [comment]
fwiw - also try to clone the repo anew, my submodule problem which resulted in build error was caused by leftover files/cache and it was just easier to clone the repo anew than to look for faulty dirs

**Comment by tarzanek** [comment]
fwiw - I will add a travis build to this repo based on this debate, so we know that the repo builds and have steps to build documented in automated CI too

**Comment by anirbanroydas** [comment]
@tarzanek  Thanks. I am going to give this a try with the above steps. Let me see if the sea monster wants me to migrate or not 😬

**Comment by anirbanroydas** [comment]
just fyi: 

Here's my `env` before starting the process;

```
LANG=C.UTF-8
XDG_SESSION_ID=88
USER=ubuntu
PWD=/home/ubuntu
HOME=/home/ubuntu
LC_CTYPE=C.UTF-8
LC_TERMINAL=iTerm2
SSH_CLIENT=<ip_address> 48600 22
LC_TERMINAL_VERSION=3.3.20190820-nightly
XDG_DATA_DIRS=/usr/local/share:/usr/share:/var/lib/snapd/desktop
SSH_TTY=/dev/pts/0
MAIL=/var/mail/ubuntu
TERM=xterm-256color
SHELL=/bin/bash
SHLVL=1
LOGNAME=ubuntu
XDG_RUNTIME_DIR=/run/user/1000
PATH=/usr/local/spark/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin
_=/usr/bin/env
```

**Comment by tarzanek** [comment]
OK, can you try having just clean PATH?
```
echo $PATH
/usr/local/bin:/bin:/sbin:/usr/bin:/usr/sbin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin
```

**Comment by tarzanek** [comment]
worst case we can try sstableloader, how big is your dataset?
Also if you are a customer, we should take it to support channel.
If you are open source user we can move this debate to open source slack - http://slack.scylladb.com/
you can find me there as Lubos

**Comment by tarzanek** [comment]
or I can send you the binary I build :-) (from official scylladb email of course)

**Comment by anirbanroydas** [comment]
@tarzanek Successfully built this time. At least the built step cleared. I will have to wait and see what happens when I start the spark job.

I did the following:
1. Deleted jdk 11
2. Installed openJDK8
3. Cleared sbt cache
4. Deleted scylla-migrator repo clone
5. Added new clone.
6. Ran build script

Few minutes later --> Build Successful. Though I saw a few Warnings. But the first hurdle is crossed.

Thanks again.

**Comment by tarzanek** [comment]
ok, so this explains what I saw too, it seems the repo is having leftovers from previous builds/jdks which later break the new build ... we should look at this

**Comment by anirbanroydas** [comment]
Btw,

I had already taken a look into sstableloader. But planned to try the migrator for the robustness it advertises. Will post here if I find anymore issues.

I will slack if needed.

**Comment by anirbanroydas** [comment]
Also once all the gaps are filled, few things that would be really useful to lay down are all the dependencies with the versions for the project, including java version, spark version, everything. The ScyllaDB blog/webinars I went over doesn't mention anything and as of now README has the spark dependency mentioned.

**Comment by tarzanek** [comment]
The versions are a little bit hidden and transitive - the magic is in https://github.com/scylladb/scylla-migrator/blob/master/build.sbt (+ plugins of course)
which specifies dependencies, but yes, we can try to point out more info (or mention this is classical sbt build, which explains where to look for what)
and yes, please do come back either here (for actual bugs, PRs, etc.) or on slack (for clarifying stuff and communication)

**Comment by anirbanroydas** [comment]
@tarzanek So I tried the migration with spark-submit but it seems to gets stuck after completing a certain number of tasks. And I keep seeing these logs

```
2019-11-12 15:01:35 INFO  StandaloneAppClient$ClientEndpoint:54 - Executor updated: app-20191112145930-0001/36 is now RUNNING
2019-11-12 15:01:35 INFO  StandaloneAppClient$ClientEndpoint:54 - Executor updated: app-20191112145930-0001/21 is now EXITED (Command exited with code 1)
2019-11-12 15:01:35 INFO  StandaloneSchedulerBackend:54 - Executor app-20191112145930-0001/21 removed: Command exited with code 1
2019-11-12 15:01:35 INFO  BlockManagerMasterEndpoint:54 - Trying to remove executor 21 from BlockManagerMaster.
2019-11-12 15:01:35 INFO  StandaloneAppClient$ClientEndpoint:54 - Executor added: app-20191112145930-0001/37 on worker-20191112140020-172.31.34.149-44027 (172.31.34.149:44027) with 2 core(s)
2019-11-12 15:01:35 INFO  BlockManagerMaster:54 - Removal of executor 21 requested
2019-11-12 15:01:35 INFO  CoarseGrainedSchedulerBackend$DriverEndpoint:54 - Asked to remove non-existent executor 21
2019-11-12 15:01:35 INFO  StandaloneSchedulerBackend:54 - Granted executor ID app-20191112145930-0001/37 on hostPort 172.31.34.149:44027 with 2 core(s), 1024.0 MB RAM
2019-11-12 15:01:35 INFO  StandaloneAppClient$ClientEndpoint:54 - Executor updated: app-20191112145930-0001/4 is now EXITED (Command exited with code 1)
2019-11-12 15:01:35 INFO  StandaloneSchedulerBackend:54 - Executor app-20191112145930-0001/4 removed: Command exited with code 1
2019-11-12 15:01:35 INFO  BlockManagerMaster:54 - Removal of executor 4 requested
2019-11-12 15:01:35 INFO  BlockManagerMasterEndpoint:54 - Trying to remove executor 4 from BlockManagerMaster.
2019-11-12 15:01:35 INFO  CoarseGrainedSchedulerBackend$DriverEndpoint:54 - Asked to remove non-existent executor 4
2019-11-12 15:01:35 INFO  StandaloneAppClient$ClientEndpoint:54 - Executor added: app-20191112145930-0001/38 on worker-20191112135915-172.31.41.51-39017 (172.31.41.51:39017) with 2 core(s)
2019-11-12 15:01:35 INFO  StandaloneSchedulerBackend:54 - Granted executor ID app-20191112145930-0001/38 on hostPort 172.31.41.51:39017 with 2 core(s), 1024.0 MB RAM
2019-11-12 15:01:35 INFO  StandaloneAppClient$ClientEndpoint:54 - Executor updated: app-20191112145930-0001/38 is now RUNNING
2019-11-12 15:01:35 INFO  StandaloneAppClient$ClientEndpoint:54 - Executor updated: app-20191112145930-0001/37 is now RUNNING
2019-11-12 15:01:35 INFO  StandaloneAppClient$ClientEndpoint:54 - Executor updated: app-20191112145930-0001/13 is now EXITED (Command exited with code 1)
2019-11-12 15:01:35 INFO  StandaloneSchedulerBackend:54 - Executor app-20191112145930-0001/13 removed: Command exited with code 1
2019-11-12 15:01:35 INFO  BlockManagerMasterEndpoint:54 - Trying to remove executor 13 from BlockManagerMaster.
2019-11-12 15:01:35 INFO  StandaloneAppClient$ClientEndpoint:54 - Executor added: app-20191112145930-0001/39 on worker-20191112135925-172.31.41.51-43939 (172.31.41.51:43939) with 2 core(s)
2019-11-12 15:01:35 INFO  BlockManagerMaster:54 - Removal of executor 13 requested
2019-11-12 15:01:35 INFO  CoarseGrainedSchedulerBackend$DriverEndpoint:54 - Asked to remove non-existent executor 13
2019-11-12 15:01:35 INFO  StandaloneSchedulerBackend:54 - Granted executor ID app-20191112145930-0001/39 on hostPort 172.31.41.51:43939 with 2 core(s), 1024.0 MB RAM
2019-11-12 15:01:35 INFO  StandaloneAppClient$ClientEndpoint:54 - Executor updated: app-20191112145930-0001/39 is now RUNNING
```
What is going wrong here?

The config.yaml
for the source I am adding one of the seed node's private IP and for the target seed node's private IP. 

the data gets migrated to new cluster to some extent which I verified by querying the new cluster for some data and number of rows written. But after the above logs start showing up in the spark master, no more data is written. 

I tried killing the job and truncating the table and starting fresh. Gave it 3 shots, but the result is the same.

**Comment by anirbanroydas** [comment]
@tarzanek  I am not sure if the above is making sense or not. Do let me know if you need more information to figure what might be going on there.

Really trying to get this tool to work.

**Comment by tarzanek** [comment]
closing, let's talk about runtime in some kind of chat, or if you found a bug let's open a new issue

**Comment by daroodar** [comment]
> @tarzanek Successfully built this time. At least the built step cleared. I will have to wait and see what happens when I start the spark job.
> 
> I did the following:
> 
> 1. Deleted jdk 11
> 2. Installed openJDK8
> 3. Cleared sbt cache
> 4. Deleted scylla-migrator repo clone
> 5. Added new clone.
> 6. Ran build script
> 
> Few minutes later --> Build Successful. Though I saw a few Warnings. But the first hurdle is crossed.
> 
> Thanks again.

Just FYI. I was stuck on the same issue and now its working after following these steps. Thanks :')
